Briefly discuss `git pull` in the code contribution workflow in the contributor guide (#2193)
PlasmaPy/PlasmaPy: docs/contributing/workflow.rst

Making a code contribution
==========================

> 

Author: ```suggestion

.. _create-branch:
```

Add and commit changes
----------------------

> Next we will go through the cycle of making changes and uploading them
> 

Author: ```suggestion
```

Next we can go through the cycle of making changes, which is usually
repeated multiple times. To get a better idea of what is being done in
each step, try running ``git status``.

#. Edit a file and save the changes.

#. In a terminal, navigate to the directory with the changed file and
   run:

   .. code-block:: bash

      git add filename

   where :samp:`{filename}` is replaced with the name of the edited

>    file(s). This step lets us line up the changes that we want to record
>    as a snapshot in history.

Author: ```suggestion
   file(s). Use ``git add *`` to add all files in the directory (except
   for files specified in :file:`.gitignore`. This step lets us line up
   the changes that we want to record as a snapshot in history.
```

   Committing a change is like preserving a snapshot of what each file
   looks like at this point in history.

   .. hint::

>       If it has been installed, |pre-commit| will perform automated
>       checks and possibly auto-fixes. If pre-commit fails, then
>       it'll be necessary to fix any remaining problems and do the
>       ``git add`` and ``git commit`` steps once more. Try using
>       ``git diff`` and ``git diff --cached`` to view the changes, and
>       :guilabel:`↑` and :guilabel:`↓` to scroll through previous
>       commands in a terminal.

Author: ```suggestion
      If |pre-commit| has been installed, it will perform automated
      checks and possibly auto-fixes. If pre-commit fails, then
      it'll be necessary to fix any remaining problems and do the
      ``git add`` and ``git commit`` steps once more. 

      Use ``git diff`` and ``git diff --cached`` to view the changes,
      and :guilabel:`↑` and :guilabel:`↓` to scroll through previous
      commands in a terminal.
```

   Committing a change is like preserving a snapshot of what each file
   looks like at this point in history.

>    .. hint::
> 
>       G

Author: ```suggestion
```

#. :ref:`Add a changelog entry <add-changelog>`, except for minor
   changes like typo fixes.

At this stage, a reviewer will perform a code review, unless it has been
marked as a draft pull request. Thank you for contributing!

Pulling changes from GitHub
---------------------------

If your branch changes on GitHub, run

.. code-block:: bash

   git pull

> to pull the changes from GitHub to your computer. If you'd like to pull
> the changes from the ``main`` branch, instead run

Author: ```suggestion
to pull the changes from GitHub to your computer.

If you'd like to pull the changes from the ``main``
branch, instead run
```

.. code-block:: bash

   git pull upstream main

If any of the changes conflict with each other, it will be necessary to
`resolve the merge conflict`_.

.. note::

      After the pull request has been created, it can be updated by
      using :bash:`git push` to update the corresponding branch on
      GitHub.

> .. note::

Author: ```suggestion
.. important::
```
